include/seastar/core/deleter.hh

```diff
@@ -126,7 +127,8 @@ deleter::~deleter() {
         std::free(to_raw_object());
         return;
     }
-    if (_impl && --_impl->refs == 0) {
+    if (_impl && _impl->refs.fetch_sub(1, std::memory_order_release) == 1) {
+        std::atomic_thread_fence(std::memory_order_acquire);
```

Comment:
Note that this is just optimization on 
```cpp
if (_impl && _impl->refs.fetch_sub(1, std::memory_order_acq_rel) == 1) {
```
that avoids the acquire fence when possible.

Reply:
I think those orderings need some comments/explanation. Or I guess reference that this just taken from a shared_ptr impl?

Reply:
> Or I guess reference that this just taken from a shared_ptr impl?

I think at least libc++ and libstdc++ just use one decrement with acq_rel rather than the fence. It is not obvious what is better. On x86 I don't think it matters. On Arm there is [a difference](https://godbolt.org/z/Ezj7qbxoM), two cheap (?) barriers vs one more expensive (?) one, not sure who wins.

Reply:
Awhile back I had read that this was how Boost's `shared_ptr` was implemented. However, looking at it now that doesn't seem to be the case and it's using acq_rel instead. I'll just switch to using that since it seems to be in-line with all current implementations of `shared_ptr`. 

I can also add a comment that the orderings are taken from those in the common implementations of `std::shared_ptr` as well if that's desirable.